Mac/MainWindow/Timeline/TimelineViewController.swift

```diff
@@ -746,6 +747,50 @@ extension TimelineViewController: NSTableViewDelegate {
 		cell.objectValue = nil
 		cell.cellData = TimelineCellData()
 	}
+
+	private func toggleArticleRead(_ article: Article) {
+		guard let undoManager = undoManager, let markUnreadCommand = MarkStatusCommand(initialArticles: [article], markingRead: !article.status.read, undoManager: undoManager) else {
+			return
+		}
+		self.runCommand(markUnreadCommand)
+	}
+
+	private func toggleArticleStarred(_ article: Article) {
+		guard let undoManager = undoManager, let markUnreadCommand = MarkStatusCommand(initialArticles: [article], markingStarred: !article.status.starred, undoManager: undoManager) else {
+			return
+		}
+		self.runCommand(markUnreadCommand)
+	}
```

Comment:
These could maybe be combined into a `toggleStatus()` function or somesuch, but I left them separate as that's how it's done in other code.